org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/ProjectConfigurationManager.java

```diff
@@ -452,6 +453,9 @@ private void updateProjectConfiguration(ProjectConfigurationRequest request, IPr
     MavenProject mavenProject = request.getMavenProject();
     Properties mavenProperties = mavenProject.getProperties();
     String sourceEncoding = mavenProperties.getProperty("project.build.sourceEncoding");
+    if(sourceEncoding == null) {
+      sourceEncoding = Charset.defaultCharset().toString();
```

Comment:
I'm missing context. What makes this change necessary fo rthe pdate?

Reply:
> I'm missing context. What makes this change necessary fo rthe pdate?

The new [Warning for projects with no explicit default encoding](https://www.eclipse.org/eclipse/news/4.24/platform.php#no-explicit-encoding-project-warning) lets some test fail because they are checking for no warnings. But due to this change there are now warnings because the projects are missing the `project.build.sourceEncoding` property.

Reply:
You may want to tweak the AbstractMavenProjectTestCase so it configure the workspace to now warn about encoding then.

Reply:
Jep, that's what I have done now. :)

Reply:
In case we later have test cases that check for that warning we may revert that and add `project.build.sourceEncoding` to those projects that require no warnings (should also be feasible).